File: task3/haskell/Sub-Task4.hs
import System.IO

printPattern :: Int -> IO ()
printPattern n = do
    let half = (n + 1) `div` 2
    mapM_ (putStrLn . pattern half) [1..half]
    mapM_ (putStrLn . pattern half) [half + 1..n]

    writeFile "output.txt" (unlines (map (pattern half) [1..half] ++ map (pattern half) [half + 1..n]))
    putStrLn "Diamond pattern has been written to output.txt"

pattern :: Int -> Int -> String
pattern half i
    | i <= half = replicate (half - i) ' ' ++ replicate (i * 2 - 1) '*'
    | otherwise = replicate (i - half) ' ' ++ replicate ((half * 2 - i) * 2 - 1) '*'

main :: IO ()
main = do
    putStrLn "Enter the number of rows: "
    n <- readLn
    printPattern n

File: task3/javascript/Sub-Task4.js
const fs = require('fs');

function printPatternToFile(n) {
    let outputFile = fs.createWriteStream('output.txt');

    for (let i = 1; i <= Math.ceil((n + 1) / 2); i++) {
        let line = ' '.repeat(Math.ceil((n + 1) / 2) - i) + '*'.repeat((i * 2) - 1);
        outputFile.write(line + '\n');
        console.log(line);
    }

    for (let i = Math.ceil((n + 1) / 2) + 1; i <= n; i++) {
        let line = ' '.repeat(i - Math.ceil((n + 1) / 2)) + '*'.repeat((n + 1 - i) * 2 - 1);
        outputFile.write(line + '\n');
        console.log(line);
    }

    outputFile.end();
    console.log("Diamond pattern has been written to output.txt");
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Enter the number of rows: ', n => {
    printPatternToFile(parseInt(n, 10));
    readline.close();
});
